File: services/client-service/src/main/java/zw/co/microbank/clientservice/service/AdminService.java
package zw.co.microbank.clientservice.service;

import zw.co.microbank.clientservice.dto.ClientResponse;

import java.util.List;

/**
 * Author: tjc
 * Created on 7/10/25
 */

public interface AdminService {
//    Admin blacklist the client
    ClientResponse blacklistClient(String clientId);

//    Admin unblacklist the client
    ClientResponse unBlacklistClient(String clientId);

//    Admin Get All client
    List<ClientResponse> getAllClients();



}

File: client/services/admin.service.ts
import { apiGet, apiPatch } from '@/lib/api'
import { ClientResponse } from '@/types'

export class AdminService {
  static async getAllClients(): Promise<ClientResponse[]> {
    return apiGet<ClientResponse[]>('/clients/admin')
  }

  static async blacklistClient(clientId: string): Promise<ClientResponse> {
    return apiPatch<ClientResponse>(`/clients/admin/${clientId}/blacklist`)
  }

  static async unblacklistClient(clientId: string): Promise<ClientResponse> {
    return apiPatch<ClientResponse>(`/clients/admin/${clientId}/unblacklist`)
  }
}